authcontroller.py

```diff
@@ -669,3 +696,74 @@ def viewmembers():
     user = current_user
     users = user.organisation.users
     return render_template("viewMembers.html", orgnisationname=user.organisation.name, users=users)
+
+
+@app.route("/updateorginfo", methods=["GET", "POST"])
+@roles_required(Role.ADMIN)
+def update_org_Info():
+    email = current_user.email
+    user = User.get(email=current_user.email, organisation=current_user.organisation)
+    form = OrgConfirmationForm()
+    redirect_uri = url_for("orcid_callback", _external=True)
+    clientSecret_url = iri_to_uri(MEMBER_API_FORM_BASE_URL) + "?" + urlencode(
+        dict(
+            new_existing=NEW_CREDENTIALS,
+            note=NOTE_ORCID + " " + user.organisation.name,
+            contact_email=email,
+            contact_name=user.name,
+            org_name=user.organisation.name,
+            cred_type=CRED_TYPE_PREMIUM,
+            app_name=APP_NAME + " at " + user.organisation.name,
+            app_description=APP_DESCRIPTION + " at " + user.organisation.name,
+            app_url=APP_URL,
+            redirect_uri_1=redirect_uri))
+
+    organisation = Organisation.get(email=email)
+    if request.method == 'POST':
```

Comment:
flask-wtf... validate_on_submit() does the both things checks the method and if it's POST, validates the form